src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -98,26 +108,28 @@ public AsymmetricSignatureProvider(AsymmetricSecurityKey key, string algorithm,
                 RSACryptoServiceProvider rsa = null;
                 if (willCreateSignatures)
                 {
-					rsa = x509Key.PrivateKey as RSACryptoServiceProvider;
+                    rsa = x509Key.PrivateKey as RSACryptoServiceProvider;
+                    rsaCryptoServiceProviderProxy = new RSACryptoServiceProviderProxy(rsa);
```

Comment:
We can code this a bit differently so that when DNXCORE50 is updated and returns rsaCNG we will not break. Basically expand the #if DNXCORE50